android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java

```diff
@@ -0,0 +1,205 @@
+import java.util.List;
+import java.util.Locale;
+
+import butterknife.BindView;
+import io.realm.Case;
+import io.realm.Realm;
+import io.realm.RealmResults;
+import timber.log.Timber;
+
+public class SearchActivity extends BaseActivity implements SearchView.OnQueryTextListener {
+
+    private GlobalSearchAdapter globalSearchAdapter;
+    private List<Object> results = new ArrayList<>();
+
+    private Realm realm = Realm.getDefaultInstance();
+
+    private SearchView searchView;
+    private String searchText;
+
+    private final String SEARCH = "SAVE_KEY_ON_ROTATE";
+
+    @BindView(R.id.search_recyclerView)
+    RecyclerView searchRecyclerView;
+    @BindView(R.id.txt_no_results)
+    TextView noResultsView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        OpenEventApp.getEventBus().register(this);
+
+        handleVisibility();
+
+        globalSearchAdapter = new GlobalSearchAdapter(results, this);
+        searchRecyclerView.setAdapter(globalSearchAdapter);
+        searchRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+
+        if (savedInstanceState != null && savedInstanceState.getString(SEARCH) != null) {
+            searchText = savedInstanceState.getString(SEARCH);
+        }
+
+        if (Intent.ACTION_SEARCH.equals(getIntent().getAction())) {
+            handleIntent(getIntent());
+        }
+
+    }
+
+    @Override
+    protected int getLayoutResource() {
+        return R.layout.activity_search;
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.menu_home, menu);
+        SearchManager searchManager =
+                (SearchManager) getSystemService(Context.SEARCH_SERVICE);
+        searchView = (SearchView) menu.findItem(R.id.action_search_home).getActionView();
+        searchView.setSearchableInfo(
+                searchManager.getSearchableInfo(getComponentName()));
+
+        searchView.setOnQueryTextListener(this);
+        if (searchText != null) {
+            searchView.setQuery(searchText, true);
+        }
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String query) {
+        if(query!=null) {
+            searchQuery(query);
+            searchText = query;
+        }
+        else{
+            results.clear();
+            handleVisibility();
+        }
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextSubmit(String query) {
+        searchView.clearFocus();
+        return true;
+    }
+
+    @Override
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        handleIntent(intent);
+    }
+
+    public void handleIntent(Intent intent) {
+        final String query = intent.getStringExtra(SearchManager.QUERY);
+        searchQuery(query);
+    }
+
+    private void searchQuery(String constraint) {
+
```

Comment:
Remove empty lines from function start. Only add to separate blocks of statements between functions